funcx_endpoint/funcx_endpoint/endpoint/rabbit_mq/task_queue_publisher.py

```diff
@@ -14,37 +16,41 @@ class TaskQueuePublisher:
 
     def __init__(
         self,
-        endpoint_uuid: str,
-        pika_conn_params: pika.connection.Parameters,
+        *,
+        endpoint_id: str,
+        conn_params: pika.connection.Parameters,
     ):
         """
         Parameters
         ----------
-        endpoint_uuid: str
+        endpoint_id: str
              Endpoint UUID string used to identify the endpoint
         pika_conn_params: pika.connection.Parameters
              Pika connection parameters to connect to RabbitMQ
         """
-        self.endpoint_uuid = endpoint_uuid
-        self.queue_name = f"{self.endpoint_uuid}.tasks"
-        self.routing_key = f"{self.endpoint_uuid}.tasks"
-        self.params = pika_conn_params
+        self.endpoint_id = endpoint_id
+        self.queue_name = f"{self.endpoint_id}.tasks"
+        self.routing_key = f"{self.endpoint_id}.tasks"
+        self.conn_params = conn_params
 
         self.exchange = "tasks"
         self.exchange_type = "direct"
 
         self._connection = None
         self._channel = None
+        # start closed ("connected" after connect)
+        self.status = RabbitPublisherStatus.closed
```

Comment:
This introduces the concept of `.status` to `TaskQueuePublisher` (perhaps more formally, anyway), but I don't see that we use it anywhere, either in the class or in the unit tests.  What is the expected or intended use of this instance variable?

Reply:
I've reviewed this a bit. As you say, `status` is not used on the publisher classes (either of them).

I believe that what happened is that there's reconnect logic on the subscriber classes which uses `self.status` and that these publisher classes did not track their own connected/disconnected status. So I added it as a kind of uniformity across all four. It might matter because the connection isn't constructed implicitly on `__init__`.

We could remove it later, but for now, I'd like to leave it in on the assumption that my past self, adding these statuses, had thought it through a bit more than I have today. It's worth knowing that this PR arose because #701 had several issues which I wanted resolved. At a certain point, I couldn't stand around waiting any longer and agreed to merge it, then immediately opened this PR with the changes that I had requested to unblock everything. So the rationale here may be buried in the dozens of comments on #701 .